akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorCompileOnlyTest.java

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.persistence.typed.javadsl;
+
+import akka.actor.Scheduler;
+import akka.actor.typed.ActorRef;
+import akka.testkit.typed.TestInbox;
+import akka.util.Timeout;
+
+import java.util.*;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import static akka.actor.typed.javadsl.AskPattern.ask;
+
+public class PersistentActorCompileOnlyTest {
+
+  public static abstract class Simple {
+    //#command
+    public static class SimpleCommand {
+      private final String data;
+
+      public SimpleCommand(String data) {
+        this.data = data;
+      }
+    }
+    //#command
+
+    //#event
+    static class SimpleEvent {
+      private final String data;
+
+      SimpleEvent(String data) {
+        this.data = data;
+      }
+    }
+    //#event
+
+    //#state
+    static class SimpleState {
+      private final List<String> events = new ArrayList<>();
+
+      SimpleState addEvent(SimpleEvent event) {
+        events.add(event.data);
```

Comment:
make a safe copy of the List and return new instance of `SimpleState`. it must be immutable

Reply:
interesting that you can access `event.data` even though it's `private`
I think those should be public anyway

Reply:
Yeah java is private to file

Reply:
I thought it was per class, but that was perhaps when all started and there were no nested classes. one per file.